packages/hdwallet-provider/src/constructor/getOptions.ts

```diff
@@ -20,29 +17,31 @@ const isPrivateKeys = (
 const isPrivateKey = (
   credentials: LegacyConstructor.Credentials
 ): credentials is PrivateKey =>
-  !isPrivateKeys(credentials) && !isMnemonicPhrase(credentials);
+  typeof credentials === "string" &&
+  credentials.length === 64 &&
+  // this is added since parseInt(mnemonic) should equal NaN and private keys
+  // should parse into a valid number
```

Comment:
Is this true? You can pass ganache a mnemonic like that looks like a private-key and it would generate priv/pub keys from that.  `ganache --wallet.mnemonic 2cf177c279d1b414ff675b914adee1349d040dd21dd6f64d335dce83da582c8a`